src/main/java/at/ac/univie/mminf/luceneSKOS/analysis/engine/jena/SKOSEngineImpl.java

```diff
@@ -227,24 +277,51 @@ public SKOSEngineImpl(InputStream inputStream, String format, List<String> langu
         searcher = new IndexSearcher(DirectoryReader.open(indexDir));
     }
 
+    private static String getHash(Reader reader) throws IOException, NoSuchAlgorithmException
+    {
+        String sha1 = "";
+        char []  cbuf = new char[1024];
+         MessageDigest crypt = MessageDigest.getInstance("SHA-1");
+        crypt.reset();
+        while (reader.read(cbuf) > 0)
+            crypt.update(new String(cbuf).getBytes(StandardCharsets.ISO_8859_1));
```

Comment:
Is there a valid reason to work with strings instead of bytes here. I don't think you need to mess with charsets. Just calculate the hash of the whole file by reading in bytes. i.e.: the sha1sum tool should get the same hash.